src/backend/Level0/CHIPBackendLevel0.hh

```diff
@@ -323,8 +352,30 @@ public:
 class CHIPContextLevel0 : public chipstar::Context {
   OpenCLFunctionInfoMap FuncInfos_;
   std::vector<LZEventPool *> EventPools_;
+  std::mutex CmdListMtx;
+  size_t NumCmdListsCreated_ = 0;
+  size_t CmdListsRequested_ = 0;
+  size_t CmdListsReused_ = 0;
+  std::stack<ze_command_list_handle_t> ZeCmdListRegStack_;
```

Comment:
I wanted to avoid this since there is a level-zero notion of an EventPool so I didn't want to confuse.

Reply:
How is level-zero's notion of object pools different to chipStar's notion?

Reply:
Event Pool is a thing that you can create in level zero. There is no such thing as an Command List Pool and I don't want to imply so. Because of that, and because it's a literal stack object, I think think ZeCmdListRegStack_ is better.

Reply:
We can go with event pool - maybe I was overthinking it.